src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/CompanyNumberValidationInterceptor.java

```diff
@@ -32,15 +35,20 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
             return false;
         }
 
-        if (companyNumber.length() != 8) {
-            ApiLogger.infoContext(reqId, "Company number length is invalid");
+        var truncatedNumber = (companyNumber.length() > MAX_COMPANY_NUMBER_LENGTH) ?
+                companyNumber.substring(0, MAX_COMPANY_NUMBER_LENGTH) : companyNumber;
```

Comment:
i think the substring length should be the same as the id's i.e. 50. Otherwise if they enter a 100 char company number then we will only log first 8 chars and that will look like it's the correct length. So we validate against 8 length, but if it is greater or invalid number, we log up to 50 chars so we can see the first 50 chars of what was entered.

Reply:
company numbers don't have 50 characters only 8 - I'm sorry but I don't understand this comment.